integrations/integrations.go

```diff
@@ -0,0 +1,246 @@
+	}
+
+	// TODO: Check if is testnet and implement `testnetify` function
+
+	app := appCreator(svrCtx.Logger, db, &fakeTraceWriter{}, svrCtx.Viper)
+	env.DeferErr("close app", app.Close)
+
+	return app, nil
+}
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L307
+// We modify this function to start a Monomer node in-process instead of a Comet node.
+func startInProcess(
+	env *environment.Env,
+	svrCtx *server.Context,
+	clientCtx *client.Context,
+	monomerCtx context.Context,
+	app servertypes.Application,
+	opts server.StartCmdOptions,
+) error {
+	svrCtx.Logger.Info("Starting Monomer node in-process")
+	err := startMonomerNode(&WrappedApplication{
+		app: app,
+	}, env, opts, monomerCtx, svrCtx)
+	if err != nil {
+		return fmt.Errorf("start Monomer node: %v", err)
+	}
+
+	if opts.PostSetup != nil {
+		var g errgroup.Group
+		if err := opts.PostSetup(svrCtx, *clientCtx, monomerCtx, &g); err != nil {
+			return fmt.Errorf("run post setup: %v", err)
+		}
+		if err := g.Wait(); err != nil {
+			return fmt.Errorf("unexpected error in PostSetup errgroup: %v", err)
+		}
+	}
+
+	return nil
+}
+
+// Starts the Monomer node in-process in place of the Comet node. The
+// `MonomerGenesisPath` flag must be set in viper before invoking this.
+func startMonomerNode(
+	wrappedApp *WrappedApplication,
+	env *environment.Env,
+	opts server.StartCmdOptions,
+	monomerCtx context.Context,
+	svrCtx *server.Context,
+) error {
+	engineWS, err := net.Listen("tcp", viper.GetString(monomerEngineWSFlag))
+	if err != nil {
+		return fmt.Errorf("create Engine websocket listener: %v", err)
+	}
+	env.DeferErr("close engine ws", engineWS.Close)
+
+	cmtListenAddr := svrCtx.Config.RPC.ListenAddress
+	cmtListenAddr = strings.TrimPrefix(cmtListenAddr, "tcp://")
+	cometListener, err := net.Listen("tcp", cmtListenAddr)
+	if err != nil {
+		return fmt.Errorf("create CometBFT listener: %v", err)
+	}
+	env.DeferErr("close comet listener", cometListener.Close)
+
+	appdb, err := opts.DBOpener(svrCtx.Config.RootDir, server.GetAppDBBackend(svrCtx.Viper))
+	if err != nil {
+		return fmt.Errorf("open app db: %v", err)
+	}
+	env.DeferErr("close app db", appdb.Close)
+
+	blockdb, err := dbm.NewDB("blockstore", dbm.BackendType(svrCtx.Config.DBBackend), svrCtx.Config.RootDir)
+	if err != nil {
+		return fmt.Errorf("create block db: %v", err)
+	}
+	env.DeferErr("close block db", blockdb.Close)
+
+	txdb, err := cometdb.NewDB("tx", cometdb.BackendType(svrCtx.Config.DBBackend), svrCtx.Config.RootDir)
+	if err != nil {
+		return fmt.Errorf("create tx db: %v", err)
+	}
+	env.DeferErr("close tx db", txdb.Close)
+
+	mempooldb, err := dbm.NewDB("mempool", dbm.BackendType(svrCtx.Config.DBBackend), svrCtx.Config.RootDir)
+	if err != nil {
+		return fmt.Errorf("create mempool db: %v", err)
+	}
+	env.DeferErr("close mempool db", mempooldb.Close)
+
+	monomerGenesisPath := viper.GetString(monomerGenesisPathFlag)
```

Comment:
I just realized we can use `svrCtx.Config.GenesisFile()` instead of adding an environment variable.